### ISSUE TITLE
All the plus signs get removed in the default.vcl

### ISSUE THREAD

**Issue by shanikasmartebusiness** [nexcess/magento-turpentine#919]
I have downloaded and updated the module from the master. After clicking 'Save Varnish Config' the plus signs get removed in  default.vcl. Please see attached images. Does anyone know how to fix it?

![version-3 vcl](https://cloud.githubusercontent.com/assets/14107584/9656012/26f54b1e-522d-11e5-8f16-5117c6fbe2fc.JPG)

and

![default vcl](https://cloud.githubusercontent.com/assets/14107584/9656011/26f0f50a-522d-11e5-8802-935f7f95aa7f.JPG)

Thank you,
Shanika

**Comment by miguelbalparda** [comment]
I don't understand your issue I see a + sign around line 71. Can you rephrase this?

**Comment by shanikasmartebusiness** [comment]
The default.vcl has got all the + signs removed after clicking 'Save Varnish Config'. Please see line 81 on the default.vcl image.

**Comment by aricwatson** [comment]
Very strange. I can't replicate this on my test system. What happens if you use the 'download varnish config' to download the generated vcl file - is it also missing all the '+' signs? If you use 'apply varnish config' does it work?

**Comment by shanikasmartebusiness** [comment]
Hi yes the downloaded vcl file also has the + signs removed. When I try to click on 'Apply Varnish Config' it gives an error, please see attached image.

![applyvarnishconfig](https://cloud.githubusercontent.com/assets/14107584/9664705/8101ea46-5263-11e5-92b9-2f3478366484.JPG)

**Comment by shanikasmartebusiness** [comment]
Any help regarding the above issue?

**Comment by aricwatson** [comment]
Sorry, but without being able to replicate this, it's very hard to diagnose. Could it be a file encoding issue? Are the plus signs missing in your version of the version-3.vcl file ?

**Comment by shanikasmartebusiness** [comment]
version-3.vcl has got the plus signs as you can see in the attached image on my 1st comment, plus signs are missing in the default.vcl file.

I have selected the Varnish option settings as shown in the image below.

![varnish](https://cloud.githubusercontent.com/assets/14107584/10019141/88643ad0-6133-11e5-9a36-1b7b2d13c968.JPG)

**Comment by shanikasmartebusiness** [comment]
Any help with this issue?

**Comment by aricwatson** [comment]
Sorry, been pretty busy - but looking at the issue again I have two ideas/comments.

First, I noticed that your generated VCL differs from the VCL template in another way - the version-3.vcl has this on line 70:
```
std.collect(req.http.cookie)
```

While your generated VCL does *not* have this code - it would be on line 81. This is odd, I'm not sure how that would disappear from the generated VCL if your Turpentine install has that template file.

Second, have you tried turning off the *strip whitespace from VCL files* option in the config? Perhaps there's some bug there that's somehow stripping out the + signs as well.

**Comment by shanikasmartebusiness** [comment]
I have tried the other 2 options in the Strip whitespace from VCL Files dropdown and I get the below errors.

![strip_whitespace_applying_directly](https://cloud.githubusercontent.com/assets/14107584/10373934/4e484622-6de8-11e5-9db5-ef4d28e8a1e9.JPG)

![strip_whitespace_never](https://cloud.githubusercontent.com/assets/14107584/10373946/59f76bec-6de8-11e5-8656-5f527ed5f721.JPG)

**Comment by abrommelhoff** [comment]
I, too, am experiencing this issue. I tried changing the offending line of code to not use "; ", which worked there, but turns out there are a number of other places in the vcl where this happens. Any word of or luck in a workaround or fix?

**Comment by abrommelhoff** [comment]
@shanikasmartebusiness If you're still encountering this error, I was able to resolve it by upgrading to the latest version (0.6.6).

**Comment by miguelbalparda** [comment]
Closing for lack of feedback.
